src/olympia/users/models.py

```diff
@@ -178,18 +178,41 @@ def __unicode__(self):
 
     @property
     def is_superuser(self):
-        return self.groups.filter(rules='*:*').exists()
+        from olympia.access import acl
+        return acl.action_allowed_user(self, amo.permissions.ADMIN)
 
     @property
     def is_staff(self):
-        from olympia.access import acl
-        return acl.action_allowed_user(self, amo.permissions.ADMIN)
+        """Property indicating whether the user should be able to log in to
+        the django admin tools. Do not use for anything else.
+
+        It only checks for whether the user is part of *any* group and that's
+        it. Having access to the tools themselves does not guarantee anything,
+        users still need to pass permission checks to view specific models and
+        modify them. In addition, everything in /admin/ is only supposed to be
+        accessible to users with VPN."""
+        return len(self.groups_list) > 0
 
     def has_perm(self, perm, obj=None):
-        return self.is_superuser
+        """Determine what the user can do in the django admin tools.
+
+        perm is in the form "<app>.<action>_<model>".
+        """
+        from olympia.access import acl
+        return acl.action_allowed_user(
+            self, amo.permissions.DJANGO_PERMISSIONS_MAPPING[perm])
 
     def has_module_perms(self, app_label):
-        return self.is_superuser
+        """Determine whether the user can see a particular app in the django
+        admin tools. """
+        # If the user has permission for any action available for any of the
+        # models of the app, or is a superuser, they can see the app in the
+        # django admin.
+        return (
+            self.is_superuser or
+            any([self.has_perm(key)
+                 for key in amo.permissions.DJANGO_PERMISSIONS_MAPPING
+                 if key.startswith('%s.' % app_label)]))
```

Comment:
`startswith` - I thought you'd listed the full `app_label`? in `DJANGO_PERMISSIONS_MAPPING`?

Reply:
keys in that dict are in the form `<app_label>.<action>_<model>` but we only receive the `app_label` part here (since it's to check whether you can access the module)